src/issuer/issuer.service.ts

```diff
@@ -32,7 +32,53 @@ export class IssuerService {
     public async issueCredential(credDefProfilePath: string, connectionId: string, entityData: any): Promise<any> {
         const [credentialData, credDefAttributes] = this.getCredDefAndSchemaData(credDefProfilePath);
         const attributes = this.formatEntityData(entityData, credDefAttributes);
-        return await this.issueCredentialSend(credentialData, connectionId, attributes);
+
+        const ret = await this.issueCredentialSend(credentialData, connectionId, attributes);
+
+        if (process.env.FLAG_RECORD_ISSUANCES === 'true') {
+            await this.recordCredential(entityData, ret);
+        }
+
+        return ret;
+    }
+
+    /**
+     * Record the issued credential using the external credential record service
+     */
+    public async recordCredential(entityData: any, issuanceRecord: any) {
+        const url = process.env.CREDENTIAL_RECORD_URL;
+        this.callService('POST', url, {
+            entityData,
+            connection_id: issuanceRecord.connection_id,
+            schema_id: issuanceRecord.schema_id,
+            credential_definition_id: issuanceRecord.credential_definition_id,
+            credential_exchange_id: issuanceRecord.credential_exchange_id,
+            state: issuanceRecord.state,
+            created_at: issuanceRecord.created_at,
+            thread_id: issuanceRecord.thread_id,
+            updated_at: issuanceRecord.updated_at,
+            credential_id: null,
+            revoc_reg_id: null,
```

Comment:
I think maybe there will be a revoc_reg_id if the cred def is set up to be revokable, perhaps we should pass through issuanceRecord.revoc_reg_id (and if it's not set it will just come back as undefined which is fine)

Reply:
gotcha, good catch.